Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -731,6 +731,8 @@ void CClientPed::Spawn(const CVector& vecPosition, float fRotation, unsigned sho
     {
         SetFrozenWaitingForGroundToLoad(true);
         m_iLoadAllModelsCounter = 10;
+
+        static_cast<CClientPlayer&>(*this).SetPlayerSpawned(true);
```

Comment:
I would avoid the expressions like this. Looks clumsy and complicated. Have you thought about moving the entire method CClientPed::Spawn under the CClientPlayer class?

Reply:
Cant move it to the player. Players inherit functions from peds and `Spawn` manage the ped's being alive. If I move it to CClientPlayer then I will break CPed

Reply:
The only place from which this method is invoked is Packet_PlayerSpawn which on its own is dedicated for players only(not peds). It cannot break non-player peds.

Reply:
> The only place from which this method is invoked is Packet_PlayerSpawn which on its own is dedicated for players only(not peds). It cannot break non-player peds.

Maybe instead of moving the entire method, I would move only `SetPlayerSpawned` there
```cpp
pPlayer->Spawn(vecPosition, fRotation, usPlayerModelID, ucInterior);
pPlayer->SetPlayerSpawned(true);
```

Reply:
That's better. But breaks the object's state coherency. Because someone in the future can invoke Spawn but forget to call SetPlayerSpawned and boom, object's state is invalid now.

Reply:
Okay, I'll move it to `CClientPlayer`

Reply:
I would suggest you to leave it as is. But after the merge create a new PR with the corresponding refactor.

Reply:
If it wont cause any issues then its better to do it now instead of waiting for it 'til merge